RIPS/rip-9998.md

```diff
@@ -0,0 +1,306 @@
+---
+eip: 0
+title: Aggregation for Native Account Abstraction
+description: Aggregation allows Smart Contract Accounts to share a single signature data between multiple AA transactions.
+author: Yoav Weiss (@yoavw), Alex Forshtat (@forshtat), Dror Tirosh (@drortirosh), Shahaf Nacson (@shahafn)
+discussions-to: https://www.google.com/
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-01
+requires: 7560
+---
+
+## Abstract
+
+RIP-7560 defined a native way to achieve Account Abstraction on Ethereum. However, one big limitation remains:
+each transaction must carry its own `signature` or other form of validation input in order to be included.
+
+We propose an extension to the RIP-7560 that introduces a validation frame that is shared by multiple transactions.
+
+This addition follows the design of [ERC-4337](./eip-4337) signature aggregation and will enable
+Ethereum transactions to natively support sharing validation inputs for the first time.
+
+## Motivation
+
+Using validation schemes that allow signature aggregation enables significant optimisations and savings on
+gas for execution and transaction data cost. This is especially relevant in the context of rollups that publish data on
+the Ethereum mainnet.
+
+Another reason to support aggregation for transactions is snark aggregation.
+Some signature schemes, such as the ones relying on `secp256r1`, are expensive to verify.
+An aggregated zero-knowledge proof that the signatures of a large number of transactions were verified
+may be cheaper to verify on-chain than verifying each of them separately.
+
+## Specification
+
+### Transaction version AA_TX_TYPE_AGGREGATED_SIG with aggregated signature data
+
+A new special version of RIP-7560 Transaction called `AA_TX_TYPE_AGGREGATED_SIG` is introduced to carry
+the aggregated signature data:
+
+```
+
+0x04 || 0x01 || rlp([chainId, transactionCount, aggregator, aggregatedSignature, aggregatedSigValidationGasLimit, accessList])
```

Comment:
I don't remember where we landed on the discussion regarding having gas fees and a signature for this transaction type.  In this form it doesn't, so it will only be created by the builder.  This precludes use cases where an aggregated batch comes from a separate entity such as an intent solver that uses `AtomicAggregator` to combine transactions from multiple users as part of the same intent.